### PR TITLE
cleaner UQ interface 

### PR DESCRIPTION
**Pull request by rjones-sandia**
A cleaner yet not clean interface between the nascent UQ code/library and NimbleSM. Still some unresolved/unresolvable issues:
- UQ uses Nimble's assembly loops
- UQ uses Nimble's BC setting routines

### CODE REVIEW

File: test/uq_notch/.git_stdout [hunk 1]
```diff
@@ -0,0 +1,51 @@
+On branch master
```
Comment (japlews) [inline test/uq_notch/.git_stdout:1]: I assume this file was accidentally added?

File: src/nimble_data_manager.h [hunk 2]
```diff
@@ -209,11 +193,6 @@ namespace nimble {
     //! Map from global node it to local node id
     std::map<int, int> global_node_id_to_local_node_id_;
 
-#ifdef NIMBLE_HAVE_UQ
-    //struct setting up the macroscale UQ model
-    UqModel uq_model_;
```
Comment (japlews) [inline src/nimble_data_manager.h:214]: Maybe make this a shared_ptr which is optionally nullptr? Avoid the need for ifdefs, just don't construct it at runtime when UQ isn't enabled.

File: src/nimble_serial.cc [hunk 3]
```diff
@@ -540,48 +498,15 @@ int ExplicitTimeIntegrator(nimble::Parser & parser,
...
-
-        if(left_sample != right_sample) { // protect against nsamples ==1
-          uq_parameters.ComputeDisplacementSensitivity(np, num_unknowns, displacement, left_off_nom_disp, right_off_nom_disp,
-                                                       displacement_sensitivities[np]);
-        }
-      }
-    }
+   // advance approximate trajectories
+   uq_model.UpdateDisplacement(delta_time);
+   uq_model.Prep();
```
Comment (hnkolla) [inline src/nimble_serial.cc:503]: Guessing that re-initialization of off nominal internal forces to 0.0 is happening in uq_model.Prep().

**Comment by rjones-sandia** [comment]
thanks Julia and Hemanth

**Pull request merged**